independent-projects/resteasy-reactive/build-support/src/main/java/org/jboss/resteasy/reactive/build/support/AllWriteableMessageBodyWriterByteBuddyPlugin.java

```diff
@@ -0,0 +1,143 @@
+     */
+    private static class MessageBodyWriterIsWriteableClassVisitor extends ClassVisitor {
+
+        private final AtomicBoolean result;
+        private Optional<AtomicBoolean> rrIsWritableResult = Optional.empty();
+        private Optional<AtomicBoolean> jaxRSIsWritableResult = Optional.empty();
+
+        private MessageBodyWriterIsWriteableClassVisitor(AtomicBoolean result) {
+            super(OpenedClassReader.ASM_API);
+            this.result = result;
+            this.result.set(false);
+        }
+
+        @Override
+        public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
+            MethodVisitor superMethodVisitor = super.visitMethod(access, name, descriptor, signature, exceptions);
+            if (name.equals("isWriteable")) {
+                // RR isWriteable
+                if ("(Ljava/lang/Class;Ljava/lang/reflect/Type;Lorg/jboss/resteasy/reactive/server/spi/ResteasyReactiveResourceInfo;Ljavax/ws/rs/core/MediaType;)Z"
+                        .equals(descriptor)) {
+                    AtomicBoolean rrResult = new AtomicBoolean(false);
+                    rrIsWritableResult = Optional.of(rrResult);
+                    return new MessageBodyWriterIsWriteableMethodVisitor(superMethodVisitor, rrResult);
+                }
+                // JAX-RS isWriteable
+                else if ("(Ljava/lang/Class;Ljava/lang/reflect/Type;[Ljava/lang/annotation/Annotation;Ljavax/ws/rs/core/MediaType;)Z"
+                        .equals(descriptor)) {
+                    AtomicBoolean standardResult = new AtomicBoolean(false);
+                    jaxRSIsWritableResult = Optional.of(standardResult);
+                    return new MessageBodyWriterIsWriteableMethodVisitor(superMethodVisitor, standardResult);
+                }
+            }
+            return superMethodVisitor;
+        }
+
+        @Override
+        public void visitEnd() {
+            if (rrIsWritableResult.isPresent()) {
+                result.set(rrIsWritableResult.get().get());
+            } else if (jaxRSIsWritableResult.isPresent()) {
+                result.set(jaxRSIsWritableResult.get().get());
+            } else {
+                result.set(false);
+            }
+            super.visitEnd();
+        }
+    }
+
+    /**
+     * This visitor sets the {@code result} to {@code true} iff the method simply does {@code return true} and nothing else
+     */
+    private static class MessageBodyWriterIsWriteableMethodVisitor extends MethodVisitor {
+
+        private final AtomicBoolean result;
+        private int insnCount = 0;
+        private boolean returnsTrue;
+
+        private MessageBodyWriterIsWriteableMethodVisitor(MethodVisitor superMethodVisitor, AtomicBoolean result) {
+            super(OpenedClassReader.ASM_API, superMethodVisitor);
+            this.result = result;
+        }
+
+        @Override
+        public void visitInsn(int opcode) {
```

Comment:
This actually only checks zero-operand instructions, no? The visitor may get called for lots of other instructions and this wouldn't detect it, no?

Reply:
That's a good point

Reply:
Fixed

Reply:
Sorry, I don't see how. The other instructions come from different `visitXXXInsn` methods, am I missing something?

Reply:
Ah, okay, now I see your point.

Reply:
Sorry, I guess I was really not clear. I'm not sure why I keep trying to make comments short instead of going into details that I know, this makes it harder for everyone because we need more back and forth. Sorry :(

Reply:
I will need https://github.com/raphw/byte-buddy/pull/1279 for this

Reply:
For now, I copied the missing class